src/server/worker/pipeline/transform/pending_job.go

```diff
@@ -108,7 +108,18 @@ func (pj *pendingJob) load() error {
 			}
 			// both commits must have succeeded - a validation error will only show up in the output
 			if metaCI.Error == "" && outputCI.Error == "" {
-				break
+				// Load the job info.
+				if _, err = pachClient.InspectJob(pj.ji.Job.Pipeline.Project.GetName(), pj.ji.Job.Pipeline.Name, metaCI.Commit.Id, false); err != nil {
+					if !errutil.IsNotFoundError(err) {
+						return err
+					}
+					// Jobs are always expected to exist. In case a pachyderm instance contains a meta commit without a job,
+					// we gracefully handle it by skipping it for base meta commit selection.
+					pj.logger.Logf("base meta commit %q could not be selected for job %q because it was missing a job was missing an associated job",
+						metaCI.Commit.String(), pj.ji.Job.String())
+				} else {
+					break
```

Comment:
just for my understanding, technically, same bug behavior could still occur here in if for some reason we fail to inspect job info?

Reply:
sort of! we would definitely fail if we couldn't inspect a job info for a meta commit. But only if the failure isn't NotFound